### ISSUE TITLE
cognitect.aws.client.api/invoke hangs when it can't find a credentials profile

### ISSUE THREAD

**Issue by dchelimsky** [cognitect-labs/aws-api#124]
## Dependencies

```clojure
{:deps {com.cognitect.aws/api       {:mvn/version "0.8.430"}
        com.cognitect.aws/endpoints {:mvn/version "1.1.11.718"}
        com.cognitect.aws/s3        {:mvn/version "784.2.593.0"}}}
```

## Description with failing test case

`cognitect.aws.client.api/invoke` hangs when it can't find a credentials profile

```clojure
(require
 '[cognitect.aws.client.api :as aws]
 '[cognitect.aws.credentials :as aws-creds])

(def s3 (aws/client {:api :s3
                     :credentials-provider 
                     (aws-creds/profile-credentials-provider "does-not-exist")}))

(aws/invoke s3 {:op :ListBuckets})
;; hangs
```

**Comment by dchelimsky** [comment]
This happens because the `credentials-profile` returns `nil` and tries and fails to put that on a channel.

**Comment by dchelimsky** [comment]
Fixed in 0.8.437
